bridge/client/app/project-board/project-board.component.ts

```diff
@@ -79,33 +86,49 @@ export class ProjectBoardComponent implements OnInit, OnDestroy {
         .pipe(
           takeUntil(this.unsubscribe$)
         ).subscribe(([traces, eventselector]: [Trace[] | undefined, string | null]) => {
-          if (traces?.length) {
-            if (eventselector) {
-              let trace = traces.find((t: Trace) => t.data.stage === eventselector && !!t.project && !!t.service);
-              if (trace) {
-                this.router.navigate(['/project', trace.project, 'sequence', trace.shkeptncontext, 'stage', trace.stage]);
-              } else {
-                trace = [...traces].reverse().find((t: Trace) => t.type === eventselector && !!t.project && !!t.service);
-                if (trace) {
-                  this.router.navigate(['/project', trace.project, 'sequence', trace.shkeptncontext, 'event', trace.id]);
-                } else {
-                  this._errorSubject.next('trace');
-                }
-              }
-            } else {
-              const trace = traces.find((t: Trace) => !!t.project && !!t.service);
-              if (trace) {
-                this.router.navigate(['/project', trace.project, 'sequence', trace.shkeptncontext]);
-              }
-            }
-          } else {
-            this._errorSubject.next('trace');
+        this.navigateToTrace(traces, eventselector);
+      });
+    }
+  }
+
+  public navigateToTrace<S, U>(traces: Trace[] | undefined, eventselector: string | null) {
```

Comment:
Are the type aliases needed here (`<S, U>`)?

Reply:
No, I don't think so. The IDE created it when automatically refactoring. I will remove it.